### PR TITLE
PAYARA-3850 Data collection API and sources

### PR DESCRIPTION
**Pull request by jbee**
### Synopsis
Introduces a new API to collect monitoring data consisting of two interface:

* `MonitoringDataSource` a `@Contract` implemented by `@Service` classes that have references to monitoring data in some form
* `MonitoringDataCollector` is passed to the sources so they can add any number of data points.

Data points are added in in a collection context that is implicit for the `MonitoringDataCollector` and can be extended using its `tag` method to create a new extended collection context.

The majority of actual code are the implementations of the `MonitoringDataSource`s where the different services add their information to the collection. 

More details can be found in the design document https://payara.atlassian.net/wiki/spaces/PAYAR/pages/870055939/Payara+Monitoring+Console+Design

### CODE REVIEW

**Comment by jbee** [comment]
jenkins test please

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java [hunk 1]
```diff
@@ -126,6 +140,80 @@ public void init() {
         }
     }
 
+    @Override
+    public void collect(MonitoringDataCollector rootCollector) {
+        MonitoringDataCollector metricsCollector = rootCollector.in("metric");
+        if (metricsEnabled != null) {
+            metricsCollector.collect("enabled", metricsEnabled);
+        }
+        if (metricsSecure != null) {
```
Comment (jGauravGupta) [inline appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: `getSecureMetrics()` is marked as deprecated, please use the `isSecurityEnabled()`:
`metricsCollector.collect("securityEnabled", isSecurityEnabled());`

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java [hunk 2]
```diff
@@ -126,6 +140,80 @@ public void init() {
         }
     }
 
+    @Override
+    public void collect(MonitoringDataCollector rootCollector) {
+        MonitoringDataCollector metricsCollector = rootCollector.in("metric");
```
Comment (jGauravGupta) [inline appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: Minor but may you name it: `metrics`

File: appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java [hunk 3]
```diff
@@ -115,44 +120,58 @@
...
                     try {
-                        String endpoint = metricsConfig.getEndpoint();
-
-                        URI remote = buildURI(server, endpoint);
-
-                        result.add(pingHealthEndpoint(remote));
-
+                        result.add(pingHealthEndpoint(instanceName,  buildURI(server, metricsConfig.getEndpoint())));
                     } catch (URISyntaxException ex) {
+                        serverHttpStatus.put(instanceName, 420);
```
Comment (jGauravGupta) [inline appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java]: Constant may be defined for `420` status code as it is not part of the HTTP standard.

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java (hunk 2 above)
Comment (jbee) [reply appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: So far I did use singular for the namespace and type tags. Overall this reads better but this is all up to discussion. I should just end up consistent.

File: appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java (hunk 3 above)
Comment (jbee) [reply appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java]: Actually I think I should change it to a 5xx code. Both cases I used it it is a server error more then a client error.

**Comment by jbee** [comment]
jenkins test please

File: appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java (hunk 3 above)
Comment (jbee) [reply appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java]: Only one really applying is plain 500 so I changed to that.

**Pull request merged**